### DISCUSSION TITLE
Slow processing

### DISCUSSION THREAD

**Discussion by FamedBear16** [brianpetro/obsidian-smart-connections#480] [General]
Hi @brianpetro 
With the last update, this plug=in has become unusable for me. I move to Open AI embedded settings and it stills freeze for a long period, before I can use obsidian.
Is there any way to revert back to the old version? I was happy with, and it was very fast.

Regards,

Pierpaolo

**Comment by brianpetro** [comment]
That shouldn't be happening. 

It would be best if you could send me a screenshot of your developer console logs to help me solve the issue.

Thanks
🌴

**Comment by FamedBear16** [comment]
Hi @brianpetro,
This is the screenshot, after that it freeze obsidian for several minutes.
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/119869050/4396d803-39f1-424a-871b-e0e446679f91)

**Comment by FamedBear16** [comment]
Hi @brianpetro,
This is the screenshot, after that it freeze obsidaoin for several minutes
my settings
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/119869050/81b7dfa8-0476-464e-bbf7-07feacdd081b)

**Comment by FamedBear16** [comment]
When i restart, it freeze again and I get this message each time
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/119869050/b21bdcc5-645e-4bff-831c-357aa935f6e3)

**Comment by brianpetro** [comment]
Thanks for sharing those.

Does the freezing end when it reaches 235/235 files?

**Reply by brianpetro** [reply]
I'm reworking some things during the "Making Smart Connections..." process that should help this.

**Reply by ru13r** [reply]
I confirm the issue. The recent version is slow and freezing Obsidian on startup. As I could see, it happens, even if it does not update the embeddings. 

It also seems at the first glance, that it has some issues with persistence. It asked to update smart_blocks embeddings several times, e. g. after restarting, each time starting the update from the beginning. Given that the commercial model is used - it comes at a price!

**Reply by brianpetro** [reply]
@FamedBear16  @ru13r `v2.0.127` addresses the slow startup and should also reduce re-embedding 🌴

**Reply by ru13r** [reply]
Great! 
Just updated to version 2.0.128. It asks to calculate the embeddings again:
![image](https://github.com/brianpetro/obsidian-smart-connections/assets/31513253/c8e62779-f475-4d5d-8c53-01a98c2240ca)
Provided that people use e.g. paid GPT 3 API, does it make any sense to force embedding refresh on each update?

**Reply by ru13r** [reply]
Slow startup is fixed. Thanks!

**Comment by FamedBear16** [comment]
Hi @brianpetro ,

It seems to be fixed now. 
I will let you know if there is any other issue.
Thanks!
